src/core/composer/qgsaddremoveitemcommand.cpp

```diff
@@ -80,5 +80,5 @@ void QgsAddRemoveItemCommand::switchState()
     emit itemAdded( mItem );
     mState = Added;
   }
-  QgsProject::instance()->setDirty( true );
+  mComposition->project()->setDirty( true );
```

Comment:
I'm wondering - should we guard all these project() uses against null pointers and allow compositions to be created without an associated project? I'm thinking here things like creating a QgsComposition from a template without it being linked to a specific project.